client/base/src/main/java/io/a2a/A2A.java

```diff
@@ -60,6 +63,108 @@ public static Message toAgentMessage(String text, String messageId) {
         return toMessage(text, Message.Role.AGENT, messageId);
     }
 
+    /**
+     * Create a user message with additional configuration options.
+     *
+     * @param text the message text
+     * @param contextId the context ID to use (optional)
+     * @param taskId the task ID to use (optional)
+     * @param parts the message parts (optional, defaults to a TextPart with the given text)
+     * @param metadata the message metadata (optional)
+     * @param referenceTaskIds the reference task IDs (optional)
+     * @return the user message
+     */
+    public static Message createUserMessage(String text, String contextId, String taskId, 
```

Comment:
Maybe instead of this method signature, we could use something like `public static Message createUserTextMessage(String text, String contextId, String taskId)` where `contextId` and `taskId` are optional to simplify the number the of method parameters here. 

This method could then call the private `toMessage` method which can be updated to add the additional parameters.